src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/ConnectionPool/HttpConnectionPool.cs

```diff
@@ -670,9 +677,10 @@ private async ValueTask<Stream> ConnectToTcpHostAsync(string host, int port, Htt
                         }
                         else
                         {
+                            IPAddress[] addresses = Dns.GetHostAddresses(host);
```

Comment:
Was the implementation already doing this exact same call elsewhere, or is this allocation new?